pkg/authorization/delegating_authorizer.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package authorization
+
+import (
+	"context"
+
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+)
+
+// NewDelegatingAuthorizer returns an authorizer that takes an authorizer
+// and delegates if the authorization decision is Allowed,
+// then returning the delegate's decision, reason, and error.
+//
+// Else, the delegate is not invoked at all
+// and the authorizer's decision (NoDecision, Denied), reason, and error is returned.
+func NewDelegatingAuthorizer(authorizer, delegate authorizer.Authorizer) authorizer.Authorizer {
```

Comment:
I was actually struggling to introduce this at all tbh as it is very little logic. I will take over your suggestion. If we find it too confusing we can inline what is done here in the authorizers (although this would be done pretty much everywhere)

Reply:
In https://github.com/kcp-dev/kcp/pull/2442 which introduced the decorators as a separate PR we decided to remove the delegating authorizer.